### PR TITLE
feat(OK): add presence

### LINKED ISSUE #6787: Ok.ru presence
### Website name

Ok.ru

### Website URL

https://ok.ru/

### Website logo

https://upload.wikimedia.org/wikipedia/commons/0/09/Ok_new_logo.png

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [ ] It is region restricted

### Description

- Being able to see what you're streaming with the thumbnail of the video displayed with the timestamp.
- And on what page you are (not that important)

### PR DESCRIPTION
**Pull request by shiseni**
## Description 
Add [OK](https://www.ok.ru/) presence.
Resolves #6787

Settings:
1. Private mode.
2. Music mode (displaying the music listening on any page).
3. Show remaining listening/viewing time.
4. Showing music covers.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image

</details>

### CODE REVIEW

File: websites/O/OK/metadata.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		"name": "SHISEN"
+	},
+	"service": "OK",
+	"description": {
+		"en": "OK is a social network service used mainly in Russia and former Soviet Republics.",
+		"ru": "ОК — социальная сеть, используемая в основном в России и бывших советских республиках."
+	},
+	"url": "ok.ru",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/Jky2SvM.png",
```
Comment (theusaf) [inline websites/O/OK/metadata.json:14]: Logos should be 512x512px at minimum. Find a larger version or upscale it using a tool like waifu2x.

File: websites/O/OK/presence.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+		]),
+		{ pathname } = document.location,
+		playMusic = document
+			.querySelector("music-mini-player")
+			?.hasAttribute("playing");
+
+	function showMusic() {
+		presenceData.details = textContent(".mini-player_name");
+		presenceData.state = textContent(".mini-player_artist");
+		presenceData.smallImageKey = playMusic ? "play" : "pause";
```
Comment (theusaf) [inline websites/O/OK/presence.ts:88]: Imgur links are preferred over Discord asset names. Also, move to an enum.

File: websites/O/OK/presence.ts [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+	return document.querySelector(tags)?.textContent;
+}
+
+function typeContent(string: string) {
+	return Assets[string as keyof typeof Assets];
+}
+
+function getMillisecondsFromString(timeString: string): number {
+	const parsedText = timeString?.split(":");
+	return (Number(parsedText[0]) * 60 + Number(parsedText[1])) * 1000;
```
Comment (theusaf) [inline websites/O/OK/presence.ts:59]: Resolve CI

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/O/OK/presence.ts (hunk 3 above)
Comment (theusaf) [inline websites/O/OK/presence.ts:59]: Resolve CI.
Possible solutions:
* Remove optional chaining on `const parsedText = ...`
* Add optional chaining to `return (Number(....`

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/O/OK/presence.ts (hunk 2 above)
Comment (shiseni) [reply websites/O/OK/presence.ts:88]: > Imgur links are preferred over Discord asset names. Also, move to an enum.

I need to load smallImageKey "play" and "pause" to imgur, right?

File: websites/O/OK/metadata.json (hunk 1 above)
Comment (shiseni) [reply websites/O/OK/metadata.json:14]: > Logos should be 512x512px at minimum. Find a larger version or upscale it using a tool like waifu2x.

Okay, I'll reupload.

File: websites/O/OK/presence.ts (hunk 3 above)
Comment (shiseni) [reply websites/O/OK/presence.ts]: > Resolve CI.

Now it is better?

Comment (theusaf) [reply websites/O/OK/presence.ts]: Looks better I think.

File: websites/O/OK/presence.ts (hunk 2 above)
Comment (theusaf) [reply websites/O/OK/presence.ts:95]: That's correct.

**Review by theusaf — APPROVED** [review]

File: websites/O/OK/iframe.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const iframe = new iFrame();
+
+iframe.on("UpdateData", async () => {
+	if (document.querySelector("video")) {
```
Comment (EncryptedDev) [inline websites/O/OK/iframe.ts:4]: This conditional is not needed - if the element doesn't exist `video` will be `false` and be falsey at the duration check.

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**